core/App/ehsm_napi.cpp

```diff
@@ -252,14 +288,21 @@ char* NAPI_Decrypt(const char* cmk,
     ehsm_data_t aad_data;
     ehsm_data_t cipher_data;
 
-    decode_cmk = base64_decode(cmk);
+    int rv = Initialize();
+    if (rv != EH_OK) {
+        retJsonObj.setCode(retJsonObj.CODE_FAILED);
+        retJsonObj.setMessage("Server exception.");
+        return retJsonObj.toChar();
+    }
+
+    decode_cmk = base64_decode(cmk_base64);
 
-    decode_cipher = base64_decode(ciphertext);
+    decode_cipher = base64_decode(ciphertext_base64);
 
     ret = ehsm_deserialize_cmk(&masterkey, (const uint8_t*)decode_cmk.data(), decode_cmk.size());
     if (ret != EH_OK) {
-        retJsonObj.code = retJsonObj.CODE_FAILED;
-        retJsonObj.msg = "Failed to deserialize cmk";
+        retJsonObj.setCode(retJsonObj.CODE_FAILED);
```

Comment:
can we set a meaningful errorcode? e.g: retJsonObj.CODE_FAILED+ret
retJsonObj.setCode(retJsonObj.CODE_FAILED);

it's will be helpful to debug the error.

Reply:
same with below

Reply:
It will be fix.